code/Controllers/CMSPageEditController.php

```diff
@@ -53,25 +53,18 @@ public function getClientConfig()
 
     /**
      * Action handler for adding pages to a campaign
-     *
-     * @param array $data
-     * @param Form $form
-     * @return DBHTMLText|HTTPResponse
      */
-    public function addtocampaign($data, $form)
+    public function addtocampaign(array $data, Form $form): HTTPResponse
     {
         $id = $data['ID'];
         $record = \Page::get()->byID($id);
 
         $handler = AddToCampaignHandler::create($this, $record);
         $results = $handler->addToCampaign($record, $data);
-        if (is_null($results)) {
-            return null;
```

Comment:
Is it safe to do this? Shouldn't we check if there's a body to the result response and return the empty response if there is no body (instead of proceeding with the below)?

Reply:
Yes it is safe

`addToCampaign()` had a strong HTTPResponse return type added - https://github.com/silverstripe/silverstripe-campaign-admin/pull/244/files#diff-005db379d07dad4da0ae604f8201249c122b800056b951f9b64b05ef977b7fdcR290

Reply:
The body might be empty, which means we shouldn't do the processing below I think. We should probably check if `$results->getBody()` is empty, and if it is, just `return $results`.

Reply:
Updated